common/azure-common-test/src/main/java/com/azure/common/test/utils/SdkContext.java

```diff
@@ -3,58 +3,151 @@
+
+import static com.azure.common.test.utils.TestConstant.DEFAULT_BUFFER_LENGTH;
 
 /**
  * The class to contain the common test methods for testing SDK.
  */
-public class SdkContext {
-    private final ResourceNamer nameGenerator;
-
-    /**
-     * Creates an SDK context that keeps track of variable names for the recorded data.
-     *
-     * <ul>
-     *     <li>If the {@code testMode} is {@link TestMode#PLAYBACK}, this will generate random variables by reading from
-     *     {@code recordedData}.</li>
-     *     <li>If the {@code testMode} is {@link TestMode#RECORD}, this will generate random variables and write them to
-     *     {@code recordedData}.</li>
-     * </ul>
-     *
-     * @param testMode The test for this context.
-     * @param recordedData The data to persist or read any variables names to or from.
-     */
-    public SdkContext(TestMode testMode, RecordedData recordedData) {
-        Objects.requireNonNull(recordedData);
-        nameGenerator =  new TestResourceNamer("", testMode, recordedData);
-    }
+public final class SdkContext {
 
     /**
      * Gets a random name.
      *
+     * @param namer the test resource random namer which is related to recorded data
      * @param prefix the prefix to be used if possible
      * @param maxLen the maximum length for the random generated name
      * @return the random name
      */
-    public String randomResourceName(String prefix, int maxLen) {
-        return nameGenerator.randomName(prefix, maxLen);
+    public static String randomTestResourceName(TestResourceNamer namer, String prefix, int maxLen) {
+        return namer.randomName(prefix, maxLen);
     }
 
     /**
-     * Generates the specified number of random resource names with the same prefix.
+     * Generates the specified number of random test resource names with the same prefix.
+     *
+     * @param namer the test resource namer with test mode and recorded data
      * @param prefix the prefix to be used if possible
      * @param maxLen the maximum length for the random generated name
      * @param count the number of names to generate
      * @return random names
      */
-    public String[] randomResourceNames(String prefix, int maxLen, int count) {
+    public static String[] randomTestResourceNames(TestResourceNamer namer, String prefix, int maxLen, int count) {
         String[] names = new String[count];
         for (int i = 0; i < count; i++) {
-            names[i] = nameGenerator.randomName(prefix, maxLen);
+            names[i] = namer.randomName(prefix, maxLen);
         }
         return names;
     }
+
+    /**
+     * Replace the text to the one defined in textReplacementRules.
+     *
+     * @param text the text which will apply the text replacement rules
+     * @param textReplacementRules the rule of text replacement
+     * @return the updated text
+     */
+    public static String applyReplacementRule(String text, Map<String, String> textReplacementRules) {
```

Comment:
Why was this moved from the PlaybackClient? I had moved the logic in there because its only usecase, from what I saw in the tests were to replace contents in a recorded HttpResponse. I don't see how this is a generic utility method.

Reply:
I saw this one appear at different places in old interceptor, so I decided to move out. I double check with RecordNetworkCallPolicy, and it does not use the same method over there. Therefore, there is no need to maek it common.  Why the RecordNetworkCallPolicy does not have the same use?

Reply:
The method signature may not be the same, but the functionality is. When you create a PlaybackClient, you pass in a set of textReplacementRules upfront, so the client applies the transformation to each response that is played back.